File: Data Structure/Array or Matrix/Medium/LET_no918_Maximum Sum Circular Subarray.py
class Solution:
    def maxSubarraySumCircular(self, nums: List[int]) -> int:
        total, cur_max, max_sum, cur_min, min_sum = 0, 0, float('-inf'), 0, float('inf')
        for num in nums:
            cur_max = max(cur_max + num, num)
            max_sum = max(max_sum, cur_max)
            cur_min = min(cur_min + num, num)
            min_sum = min(min_sum, cur_min)
            total += num
        return max(max_sum, total - min_sum) if min_sum != total else max_sum

File: Data Structure/Array or Matrix/Medium/LET_no918_Maximum Sum Circular Subarray.cpp
class Solution {
public:
    int maxSubarraySumCircular(vector<int>& nums) {
        int cur_min = 0, cur_max = 0;
        int total_min = INT_MAX, total_max = INT_MIN, total_sum = 0;
        for (int num : nums) {
            cur_max = max(cur_max + num, num);
            total_max = max(total_max, cur_max);
            cur_min = min(cur_min + num, num);
            total_min = min(total_min, cur_min);
            total_sum += num;
        }

        return total_sum != total_min ? max(total_max, total_sum - total_min) : total_max;
    }
};